canonical-web-and-design/maas-ui
# Build add machine form.

## Done
- Built "Add machine" form. Extra MACs and Save and add another buttons to be implemented in #794 and #795 

## QA
- Go to /r/machines/add and check that the add machine form shows up.
- Check that the Minimum kernel field is correctly set to the default if that has been set - otherwise it should be "No minimum kernel"
- Check that the MAC address field validates correctly
- Choose different power types from the select and check that each field is validated correctly
- Check that you can add a machine if all the required fields have been filled. For simplicity's sake you can choose "Manual" from the Power type dropdown because it has no extra power parameter fields.
- Check that you are redirected to /r/machines with a notification confirming the machine was added.
- Go back to the form and enter a MAC address you know is already used. Submit the form and check that a server error displays at the top of the form.

Fixes canonical-web-and-design/MAAS-squad#1729

## Screenshots
### Default
![screenshot_25](https://user-images.githubusercontent.com/25733845/74458180-9d041d00-4e89-11ea-92e3-21648c2d0a27.png)

### Client-side validation
![screenshot_26](https://user-images.githubusercontent.com/25733845/74458190-9fff0d80-4e89-11ea-8b32-49f37bc81b74.png)

### Server errors
![screenshot_28](https://user-images.githubusercontent.com/25733845/74460290-f1f56280-4e8c-11ea-86fb-b2109692b188.png)
